src/client.rs

```diff
@@ -87,8 +89,12 @@ pub struct Tidal {
 impl Tidal {
     #[must_use]
     pub fn new(credentials: TidalCredentials) -> Self {
-        if credentials.to_owned().session_info.unwrap().session_id.is_none() {
-            panic!("You need an authenticated credential to use Tidal");
+        let _sesion_id = match &credentials.session_info {
+            None => panic!("A session needs to be obtatined before using Tidal"),
+            Some(session_info) => match &session_info.session_id {
+                Some(session_id) => session_id,
+                None => panic!("You need an authenticated credential to use Tidal"),
```

Comment:
Here maybe reduce this expresion to `session_info.session_id.expect("You need an authenticated credential to use Tidal") and then you don't need to nest 2 `match` blocks

Reply:
good one 👍